src/main/scala/com/fulcrumgenomics/alignment/Alignment.scala

```diff
@@ -107,10 +113,10 @@ case class Cigar(elems: IndexedSeq[CigarElem]) extends Iterable[CigarElem] {
   def reverseIterator: Iterator[CigarElem] = elems.reverseIterator
 
   /** Returns the length of the alignment on the query sequence. */
-  def lengthOnQuery: Int = elems.filter(_.operator.consumesReadBases()).map(_.length).sum
+  def lengthOnQuery: Int = elems.foldLeft(0)((sum, elem) => sum + elem.lengthOnQuery)
```

Comment:
It's more that I expect this may get called quite a bit, and I figured that a fold here would be more efficient than the `.map().sum` which would presumably a) create an intermediate `Seq[Int]` and also box all the values along the way.